qiskit/providers/aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,204 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Statevector quantum state class.
+"""
+import copy
+import numpy as np
+
+from qiskit.quantum_info.states import Statevector
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.providers.aer.quantum_info.states.aer_state import AerState
+
+
+class AerStatevector(Statevector):
```

Comment:
Did you want to have a method to do the expectation value calculation more efficiently in the C++ code. Terra's version is partially in rust and multithreaded (and the rest is in numpy) but I feel like aer state can compute that faster directly.
Similarly what about `probabilities()`, I feel like we can do the sampling much more efficiently on the aer state.

Also some of the mutation methods like `reset()` I feel like we need overloaded versions of here otherwise won't `_aer_state` get out of sync with data?